org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java

```diff
@@ -1263,6 +1271,8 @@ public static IrritantSet warningTokenToIrritants(String warningToken) {
 			case 'r' :
 				if ("rawtypes".equals(warningToken)) //$NON-NLS-1$
 					return IrritantSet.RAW;
+				if ("reference-comparison".equals(warningToken)) //$NON-NLS-1$
+					return IrritantSet.UNLIKELY_ARGUMENT_TYPE;
```

Comment:
We shouldn't share `UNLIKELY_ARGUMENT_TYPE` here, since warning tokens (as used for `@SW`) should map to distinct irritant sets. This implies we'll need a new IrritantSet constant for the new token.